services/frontend-service/src/ui/components/ServiceLane/ServiceLane.tsx

```diff
@@ -22,6 +22,25 @@ import { Tooltip } from '@material-ui/core';
 import { useNavigate } from 'react-router-dom';
 import * as React from 'react';
 
+// number of releases on home. based on design
+const numberOfDisplayedReleasesOnHome = 6;
+
+const getAllDisplayedReleases = (deployedReleases: number[], allReleases: number[]): number[] => {
+    // number of remaining releases to get from history
+    const numOfTrailingReleases = numberOfDisplayedReleasesOnHome - deployedReleases.length;
+    // last deployed release e.g. Prod
+    const oldestDeployedRelease = deployedReleases[deployedReleases.length - 1];
+
+    const allDisplayedReleases = deployedReleases;
+    // go over the remaining spots and fill from history
+    for (let i = 1; i <= numOfTrailingReleases; i++) {
+        const index = allReleases.indexOf(oldestDeployedRelease) + i;
+        if (index >= allReleases.length) break;
+        allDisplayedReleases.push(allReleases[index]);
+    }
+    return allDisplayedReleases;
+};
```

Comment:
I can't help but think this is more complicated than it needs to be.
Can't we just do
`(deployedRelease + allRelease).slice(0, numberOfDisplayedReleasesOnHome)`
?

Reply:
The logic has to stay the same because:
* `deployedReleases` is definitely contained in `allReleases`
* adding them up would lead to repetition.
* We need to throw away all versions newer than the oldest deployed one.
  e.g. we don't display ver. 6 before ver. 4 which is deployed to production. 
  the important/ deployed releases should go first and then we fill from history.
 
 -----
 
 I did improve it and use slice instead of a for loop. :+1: